dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java

```diff
@@ -266,10 +266,39 @@ private void writeField(FieldDef fd, Object fieldValue) throws Exception
         {
             case BINARY:
             {
-                byte[] data = fieldValue==null?new byte[0]:(byte[]) fieldValue;
-                long dataSize = data.length;
-                writeUnsigned(dataSize);
-                writeByteArray(data);
+                if (fd.isFixed())
+                {
+                    byte[] data = fieldValue==null?new byte[0]:(byte[]) fieldValue;
+                    int bytesToWrite = data.length;
+                    if (bytesToWrite > fd.getDataLen())
+                    {
+                        bytesToWrite = (int) fd.getDataLen();
```

Comment:
some data is being cut off? does it make sense to log?

Reply:
Good call, I will add that in